sql/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -130,9 +134,24 @@ private void expandColumn(AnalyzedColumnDefinition column) {
         }
     }
 
-    public List<String> primaryKeys() {
+    public Set<String> notNullColumns() {
+        if (notNullColumns == null) {
+            notNullColumns = new LinkedHashSet<>();
+            for (AnalyzedColumnDefinition column : columns) {
+                String fqn = column.ident().fqn();
+                if (column.isNotNull() && !primaryKeys().contains(fqn)) {
+                    if (!notNullColumns.contains(fqn)) {
```

Comment:
My logic behind the check is to keep the order of the columns in the notNullColumns set, but maybe we don't need to keep...